### PR TITLE
fix: resource `aws_cloudwatch_log_group` should respect the context.enabled flag

### PR DESCRIPTION
**Pull request by bartelemi**
## what
* Check the context flag as well as the cloudwatch_log_group_enabled flag to provision CW log group.
* Additionally, add the same check before calls to coalesce functions that rely on null-label module. If the context is disabled, the label module will return null and call to `coalesce` will fail.

## why
* To destroy/disable this module we have to set two flags when only one should be enough

## references
* #181

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ module "ecr" {
 }
 
 resource "aws_cloudwatch_log_group" "app" {
-  count = var.cloudwatch_log_group_enabled ? 1 : 0
+  count = module.this.enabled && var.cloudwatch_log_group_enabled ? 1 : 0
```
[the remark this answers is not in the document]
Comment (bartelemi) [reply main.tf:16]: Not really relevant to this PR.

File: main.tf [hunk 2]
```diff
@@ -104,13 +104,13 @@ module "container_definition" {
 
 locals {
   alb = {
-    container_name   = coalesce(var.alb_container_name, module.this.id)
+    container_name   = module.this.enabled ? coalesce(var.alb_container_name, module.this.id) : null
     container_port   = var.container_port
```
Comment (nitrocode) [inline main.tf:108]: Why do the container names also need to use the enable logic

Comment (bartelemi) [reply main.tf:108]: AFAIK `coalesce` has to return a not-null value and if this module is disabled and `var.alb_container_name` is not set (it's optional), this line will raise an error.

I could replace it with `try` if you don't like this approach:
```tf
container_name = try(coalesce(var.alb_container_name, module.this.id), null)
```

Comment (nitrocode) [reply main.tf:108]: Can we update the line to drop coalesce then?

```
    container_name   = var.alb_container_name != null ? var.alb_container_name : module.this.id
```

Comment (bartelemi) [reply main.tf:108]: Done 👍🏻

**Review by Gowiem — APPROVED** [review]
Seems simple enough to me and you've addressed some previous feedback 👍

**Comment by Gowiem** [comment]
@bartelemi can you run the following and push your changes?

```
make init
make github/init
make readme
```

**Review by Gowiem — APPROVED** [review]

**Pull request merged**